websites/serializers.py

```diff
@@ -41,14 +41,66 @@
 from websites.models import Website, WebsiteContent, WebsiteStarter
 from websites.permissions import is_global_admin, is_site_admin
 from websites.site_config_api import SiteConfig
-from websites.utils import permissions_group_name_for_role
+from websites.utils import (
+    get_dict_field,
+    permissions_group_name_for_role,
+    set_dict_field,
+)
 
 log = logging.getLogger(__name__)
 
 
 ROLE_ERROR_MESSAGES = {"invalid_choice": "Invalid role", "required": "Role is required"}
 
 
+RELATION_URL_FIELDS = (
+    ("video_files.video_captions_resource", settings.YT_FIELD_CAPTIONS),
+    ("video_files.video_transcript_resource", settings.YT_FIELD_TRANSCRIPT),
+)
+
+
+def relation_value_to_id(value) -> str | None:
+    if not value:
+        return None
+    if isinstance(value, dict):
+        return value.get("content") or None
+    if isinstance(value, list):
+        first = value[0] if value else None
+        return first[0] if isinstance(first, list | tuple) else first
+    return value if isinstance(value, str) else None
+
+
+def resource_file_path(resource: WebsiteContent | None) -> str | None:
+    if not resource:
+        return None
+    if getattr(resource, "file", None) and getattr(resource.file, "name", None):
```

Comment:
could use hasattr i think

Reply:
Simplified, although still using `getattr`, in 922d4b04c94b5bcccbf3277358c5b4726e42dd2d.